store/store.go

```diff
@@ -52,7 +52,17 @@ type RootStore interface {
 	// TracingEnabled returns true if tracing is enabled on the RootStore.
 	TracingEnabled() bool
 
+	// LoadVersion loads the RootStore to the given version.
 	LoadVersion(version uint64) error
+	// LoadVersionAndUpgrade behaves identically to LoadVersion except it also
+	// accepts a StoreUpgrades object that defines a series of transformations to
+	// apply to store keys (if any).
+	//
+	// Note, handling StoreUpgrades is optional depending on the underlying RootStore
+	// implementation.
+	LoadVersionAndUpgrade(version uint64, upgrades *StoreUpgrades) error
```

Comment:
should this be a separate interface that we cast a store to? if its optional then its more boiler plate for implementors

Reply:
It's more so that the use of the `upgrades` argument is optional. In such a case, `LoadVersionAndUpgrade` would behave the same way `LoadVersion` does.

We can move this to a separate interface if you'd like?

Reply:
i dont think it would make such a big ux win, this is an area where lots of changes wont happen

Reply:
So move that method to an interface?

Reply:
I am with @tac0turtle that we should move that method to an optional interface perhaps called "UpgradeableStore" or "StoreWithUpgrades" say
```go
type UpgradeableStore inteface {
   Store
   LoadVersionAndUpgrade(version uint64, upgrades *StoreUpgrades) error
}
```